src/main/java/io/pivotal/cfapp/controller/LegacyWorkloadsController.java

```diff
@@ -28,16 +36,24 @@ public LegacyWorkloadsController(
         this.util = new TkServiceUtil(tkService);
     }
 
+
     @GetMapping(value = { "/snapshot/detail/legacy" } )
-	public Mono<ResponseEntity<Workloads>> getLegacyWorkloads(@RequestParam("stacks") String stacks) {
+    public Mono<ResponseEntity<Workloads>> getLegacyWorkloads(@RequestParam(value = "stacks", defaultValue = "", required = false) String stacks,
```

Comment:
What happens when both `stacks` and `serviceOfferings` are (`""`) or not defined?  Seems like this should be an exception and return a 400 (bad request). `Stacks` had been required before we added `serviceOfferings`.

Reply:
Nevermind, looks like you got it covered in `PoliciesValidator`.